initfiles/bin/init_thorslave.in

```diff
@@ -44,24 +44,14 @@ slavename=thorslave_${hpcc_compname}
 stop_slaves()
 {
     local timer=15
-    local isAlive=0
-
-    log "Attempting to kill $slavename with SIGTERM"
-    killall -SIGTERM $slavename > /dev/null 2>&1
-    while [[ $isAlive -eq 0 && $timer -gt 0 ]];do
-        killall -0 $slavename > /dev/null 2>&1
-        isAlive=$?
-        [[ $isAlive -eq 0 ]] && sleep 0.5
-        ((timer--))
+    local _pidarray
+    local _pidname
+    
+    _pidarray=($( ls -l ${PID_DIR} | awk -v regpattern=${slavename}_[1-9][0-9]*\.pid '$0~regpattern {print $9;}' ))
```

Comment:
This uses the regex you were using in the pid file removal in line 64.  It should match exactly to the thorslave_compname_slavenum, even if compname has  underscores and digits.

Reply:
Use ls -1 (a digit - the number one, not the letter l) so that the listing is just filenames.

Reply:
It's a an -l on my screen. The top of it doesn't have the downward slant that the 1 in the regex has.

Reply:
Right.  Please use ls -1